### PR TITLE
Changes For Enable Remote  Debugging

### PR DESCRIPTION
**Pull request by Ishad-M-I-M**
Updated instructions in the jasminegraph documentation. https://docs.google.com/document/d/1j0UdxZVwPv4f13V9muCnWQN_m1zYO_3QNIfZ6Bt8Mk0

### CODE REVIEW

File: run-docker.sh [hunk 1]
```diff
@@ -60,10 +61,18 @@ else
...
 
 export LD_LIBRARY_PATH=/usr/local/lib
-if [ $MODE -eq 1 ]; then
-    ./JasmineGraph "docker" $MODE $MASTERIP $WORKERS $WORKERIP $ENABLE_NMON
+if [ -n "$DEBUG" ]; then
+  if [ $MODE -eq 1 ]; then
+      gdbserver 0.0.0.0:$DEBUG ./JasmineGraph "docker" $MODE $MASTERIP $WORKERS $WORKERIP $ENABLE_NMON
+  else
+      gdbserver 0.0.0.0:$DEBUG ./JasmineGraph "docker" $MODE $HOST_NAME $MASTERIP $SERVER_PORT $SERVER_DATA_PORT $ENABLE_NMON
+  fi;
```
Comment (thevindu-w) [inline run-docker.sh:69]:
```suggestion
    if [ $MODE -eq 1 ]; then
        gdbserver 0.0.0.0:$DEBUG ./JasmineGraph "docker" $MODE $MASTERIP $WORKERS $WORKERIP $ENABLE_NMON
    else
        gdbserver 0.0.0.0:$DEBUG ./JasmineGraph "docker" $MODE $HOST_NAME $MASTERIP $SERVER_PORT $SERVER_DATA_PORT $ENABLE_NMON
    fi
```

File: run-docker.sh [hunk 2]
```diff
@@ -60,10 +61,18 @@ else
...
 else
-    ./JasmineGraph "docker" $MODE $HOST_NAME $MASTERIP $SERVER_PORT $SERVER_DATA_PORT $ENABLE_NMON
+  if [ $MODE -eq 1 ]; then
+      ./JasmineGraph "docker" $MODE $MASTERIP $WORKERS $WORKERIP $ENABLE_NMON
+  else
+      ./JasmineGraph "docker" $MODE $HOST_NAME $MASTERIP $SERVER_PORT $SERVER_DATA_PORT $ENABLE_NMON
+  fi;
```
Comment (thevindu-w) [inline run-docker.sh:75]:
```suggestion
    if [ $MODE -eq 1 ]; then
        ./JasmineGraph "docker" $MODE $MASTERIP $WORKERS $WORKERIP $ENABLE_NMON
    else
        ./JasmineGraph "docker" $MODE $HOST_NAME $MASTERIP $SERVER_PORT $SERVER_DATA_PORT $ENABLE_NMON
    fi
```

**Review by thevindu-w — COMMENTED** [review]
For shfmt lint failure

File: Dockerfile [hunk 3]
```diff
@@ -4,6 +4,9 @@ ENV JASMINEGRAPH_HOME="/home/ubuntu/software/jasminegraph"
 
 WORKDIR /home/ubuntu/software/jasminegraph
 
+ARG DEBUG="false"
+RUN if [ "$DEBUG" = "true" ]; then apt-get update && apt-get install --no-install-recommends -y gdb gdbserver ; fi
```
Comment (Ishad-M-I-M) [inline Dockerfile:8]:
```suggestion
RUN if [ "$DEBUG" = "true" ]; then apt-get update \
&& apt-get install --no-install-recommends -y gdb gdbserver \
&& apt-get clean; fi
```

**Review by miyurud — APPROVED** [review]

**Pull request merged**